### PR TITLE
Add a canvas example

### LINKED ISSUE #24: Examples for using DrawContext
I can't seem to comprehend how I'm supposed to use the DrawContext (or where do I get uiDrawContext ptr for instancing new DrawContext). Quick example could help

### PR DESCRIPTION
**Pull request by barzamin**
fixes #24. the example just renders the following:

/cc @Pomettini (you wanted to know about this ~12 days ago. sorry! hopefully this is a good reference)

### CODE REVIEW

**Comment by pomettini** [comment]
Thanks a lot for remembering, @barzamin, much appreciated! Will try it right away! 😄

**Comment by barzamin** [comment]
issue on win systems: `uiDrawFillModeWinding` is an enum value of type `c_int`; I think some MSVC typedefs don't match the way stuff works on *nix. best possible fix: stop using raw enum values from the `-sys` crate and actually write a draw mode enum.

```
error[E0308]: mismatched types
   --> iui\examples\canvas.rs:17:35
    |
 17 |         let path = Path::new(ctx, uiDrawFillModeWinding);
    |                                   ^^^^^^^^^^^^^^^^^^^^^ expected u32, found i32
 
error[E0308]: mismatched types
   --> iui\examples\canvas.rs:30:35
    |
 30 |         let path = Path::new(ctx, uiDrawFillModeWinding);
    |                                   ^^^^^^^^^^^^^^^^^^^^^ expected u32, found i32
```

File: iui/src/draw/path.rs [hunk 1]
```diff
@@ -14,22 +12,37 @@ impl Drop for Path {
...
+    Winding,
+    Alternate,
+}
+
+impl FillMode {
+    fn into_ui_fillmode(self) -> uiDrawFillMode {
+        match self {
+            FillMode::Winding => uiDrawFillModeWinding,
+            FillMode::Alternate => uiDrawFillModeAlternate,
+        }
```
Comment (NoraCodes) [inline iui/src/draw/path.rs]: Perhaps add a cast here?

```suggestion
        } as uiDrawFillMode
```

This is how I solved the platform-dependent type error in 27312b4

File: iui/src/draw/path.rs [hunk 2]
```diff
@@ -14,22 +12,37 @@ impl Drop for Path {
     }
 }
 
+#[derive(Clone, Copy, PartialEq)]
+pub enum FillMode {
```
Comment (NoraCodes) [inline iui/src/draw/path.rs:17]: Could you add a doc comment onto here? Aside from this the PR looks good, thanks so much!

**Review by NoraCodes — DISMISSED** [review]

**Comment by barzamin** [comment]
Sure, I can totally do that.

**Comment by barzamin** [comment]
cc @NoraCodes

**Comment by NoraCodes** [comment]
Awesome, thank you!

**Pull request merged**